### ISSUE TITLE
[7.6.33] parse.cpp:904:5: warning: '%s' directive output may be truncated writing up to 4095 bytes into a region of size 254

### ISSUE THREAD

**Issue by Germano0** [BOINC/boinc#1936]
I got the following error while compiling BOINC 7.6.33 in Fedora

```
parse.cpp: In member function 'int XML_PARSER::copy_element(std::__cxx11::string&)':
parse.cpp:904:5: warning: '%s' directive output may be truncated writing up to 4095 bytes into a region of size 254 [-Wformat-truncation=]
 int XML_PARSER::copy_element(string& out) {
     ^~~~~~~~~~
In file included from /usr/include/stdio.h:939:0,
                 from /usr/include/c++/7/cstdio:42,
                 from /usr/include/c++/7/ext/string_conversions.h:43,
                 from /usr/include/c++/7/bits/basic_string.h:6159,
                 from /usr/include/c++/7/string:52,
                 from parse.cpp:33:
/usr/include/bits/stdio2.h:65:44: note: '__builtin___snprintf_chk' output between 4 and 4099 bytes into a destination of size 256
        __bos (__s), __fmt, __va_arg_pack ());
```

**Comment by davidpanderson** [comment]
Thanks; I checked in a fix.
-- David

**Comment by ChristianBeer** [comment]
Fixed in d11e7752756bb5773a644fd1dd4dec8b7bb9caf2
